lib/AmdMainTemplatePlugin.js

```diff
@@ -1,43 +1,55 @@
 /*
-	MIT License http://www.opensource.org/licenses/mit-license.php
-	Author Tobias Koppers @sokra
-*/
-var ConcatSource = require("webpack-sources").ConcatSource;
+ MIT License http://www.opensource.org/licenses/mit-license.php
+ Author Tobias Koppers @sokra
+ */
 
-function AmdMainTemplatePlugin(name) {
-	this.name = name;
+"use strict";
+
+const ConcatSource = require("webpack-sources").ConcatSource;
+
+class AmdMainTemplatePlugin {
+	constructor(name) {
+		this.name = name;
+	}
+
+	apply(compilation) {
+		const mainTemplate = compilation.mainTemplate;
```

Comment:
Why is this the case, btw? `const` means it can't be reassigned, not that it can't be modified; that'd be `Object.freeze`.